check.py

```diff
@@ -47,10 +54,55 @@ def warn(text):
 
 # setup
 
-BASE_DIR = os.path.abspath('test')
-WATERFALL_BUILD_DIR = os.path.join(BASE_DIR, 'wasm-install')
+# Locate Binaryen build artifacts directory (bin/ by default)
+if not options.binaryen_bin:
+  if os.environ.get('BINARYEN_ROOT'):
+    if os.path.isdir(os.path.join(os.environ.get('BINARYEN_ROOT'), 'bin')): options.binaryen_bin = os.path.join(os.environ.get('BINARYEN_ROOT'), 'bin')
+    else: options.binaryen_bin = os.environ.get('BINARYEN_ROOT')
+  else:
+    options.binaryen_bin = 'bin'
+
+if not os.path.isfile(os.path.join(options.binaryen_bin, 'wasm-dis')) and not os.path.isfile(os.path.join(options.binaryen_bin, 'wasm-dis.exe')):
+  warn('Binaryen not found (or has not been successfully built to bin/ ?')
+
+# Locate Binaryen source directory if not specified.
+if not options.binaryen_root:
+  options.binaryen_root = os.path.dirname(os.path.abspath(__file__))
+
+options.binaryen_test = os.path.join(options.binaryen_root, 'test')
+
+# Finds the given executable 'program' in PATH. Operates like the Unix tool 'which'.
+def which(program):
+  def is_exe(fpath): return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+  fpath, fname = os.path.split(program)
+  if fpath:
+    if is_exe(program): return program
+  else:
+    for path in os.environ["PATH"].split(os.pathsep):
+      path = path.strip('"')
+      exe_file = os.path.join(path, program)
+      if is_exe(exe_file): return exe_file
+      if not '.' in fname:
+        if is_exe(exe_file + '.exe'): return exe_file + '.exe'
+        if is_exe(exe_file + '.cmd'): return exe_file + '.cmd'
+        if is_exe(exe_file + '.bat'): return exe_file + '.bat'
+
+WATERFALL_BUILD_DIR = os.path.join(options.binaryen_test, 'wasm-install')
 BIN_DIR = os.path.abspath(os.path.join(WATERFALL_BUILD_DIR, 'wasm-install', 'bin'))
 
+GCC = os.environ.get('CC') or which('mingw32-gcc') or which('gcc') or ''
```

Comment:
How about we add clang here, so maybe this will work on mac too? And I guess if we're going to do that we could call it `HOSTCC` or `NATIVECC` or something like that instead, since it doesn't matter that it's gcc, just that it's a compiler that targets the host/build machine rather than wasm.